ui/static/ui/js/utils.jsx

```diff
@@ -116,6 +116,109 @@ define("utils", ["jquery", "lodash", "react", "react_infinite", "select2"],
+   *
+   * @param {String} id - Bootstrap dialog id
+   * @param {String} actionButtonName - Name of action button can be any of delete, remove, ok etc
+   * @param {String} actionButtonClass - Class(es) for action button
+   * @param {String} title - Title/heading of confirmation dialog
+   * @param {String} message - Content of dialog
+   * @param {confirmationResponse} confirmationSuccess - Call back that sends confirmation status
+  */
+  var ConfirmationDialog =  React.createClass({
+    getInitialState: function() {
+      return {
+        id: 'confirm-box',
+        className: 'modal fade',
+        actionButtonName: 'Confirm',
+        actionButtonClass: 'btn btn-ok'
+      };
+    },
+    componentWillMount: function() {
+      if (this.props.actionButtonName) {
+        this.setState({actionButtonName: this.props.actionButtonName});
+      }
+
+      if (this.props.actionButtonClass) {
+        this.setState({actionButtonClass: this.props.actionButtonClass});
+      }
+
+      if (this.props.id) {
+        this.setState({id: this.props.id});
+      }
+
+      if (this.props.className) {
+        this.setState({className: this.props.className});
+      }
+    },
+    render: function() {
+      if (!this.props || !this.props.message) {
+        return null;
+      }
+      var title = '';
+      if (this.props.title) {
+        title = <h4 className="modal-title">{this.props.title}</h4>;
+      }
+
+      return (
+        <div className={this.state.className} id={this.state.id}>
+          <div className="modal-dialog">
+            <div className="modal-content">
+              <div className="modal-header">
+                <button type="button" className="close"
+                   data-dismiss="modal"><i className="fa fa-remove"/></button>
+                {title}
+              </div>
+              <div className="modal-body">
+                <p>{this.props.message}</p>
+              </div>
+              <div className="modal-footer">
+                <button type="button" className="btn btn-default"
+                  data-dismiss="modal"
+                  onClick={this.confirmationFailure}>Cancel</button>
+                <button className={this.state.actionButtonClass}
+                  data-dismiss="modal"
+                  onClick={this.confirmationSuccess}
+                >{this.state.actionButtonName}</button>
+              </div>
+            </div>
+          </div>
+        </div>
+      );
+    },
+    componentDidMount : function() {
+      $("#" + this.state.id).modal('show');
+    },
+    confirmationSuccess: function() {
+      if (this.props.tag) {
+        this.props.confirmationSuccess(true, this.props.tag);
+      } else {
+        this.props.confirmationSuccess(true);
+      }
+    },
+    confirmationFailure: function() {
+      if (this.props.tag) {
+        this.props.confirmationSuccess(false, this.props.tag);
+      } else {
+        this.props.confirmationSuccess(false);
```

Comment:
The second argument will be passed `undefined` anyway, the if statement seems unnecessary here and above in `confirmationSuccess`